pagopa/io-app
# feat: [IOPID-2026] Android `minSdk 25` banner

## Short description
This PR adds an informational banner to notify users running the app on Android devices with OS versions earlier than `7.1` ([API Level 25](https://developer.android.com/tools/releases/platforms#7.1)) that their devices will soon no longer be able to install App IO..

> [!note]
> - screenshots has `7.1.1` instead of `7.1` because they've been taken before changing the copy.
> - the banner functional component has been placed inside `LandingScree` without creating a separate file, because it's a logic that must be removed in the near future.

<details><summary>Small screen</summary>
<p>

| Banner only | Banner along with status banner |
| - | - |
| <img src="https://github.com/user-attachments/assets/698d22c6-c563-4b3c-9d46-e7726c929389" /> | <img src="https://github.com/user-attachments/assets/d2bef0ac-691d-4ff3-b170-f5442372b401" /> |
| <img src="https://github.com/user-attachments/assets/5dfa183a-b588-4d5f-b1e3-274d2974978a" /> | <img src="https://github.com/user-attachments/assets/924ef938-a125-45fc-8955-70645d6d91b5" /> |
| <img src="https://github.com/user-attachments/assets/17e3bc69-a690-452a-889d-ec1fc72b6929" /> | <img src="https://github.com/user-attachments/assets/4ebe504a-b73e-40bd-a5d3-ce093cadf0c3" /> |

</p>
</details> 

<details><summary>Regular screen</summary>
<p>

| Banner only | Banner along with status banner |
| - | - |
| <img src="https://github.com/user-attachments/assets/2ca7f9de-065f-485e-8870-a389a33ff552" /> | <img src="https://github.com/user-attachments/assets/9ea2455c-0ee2-4939-87fa-406ceae28fdd" /> |
| <img src="https://github.com/user-attachments/assets/2d1d5e62-9d47-49b0-9e0c-2b3cf7bac45c" /> | <img src="https://github.com/user-attachments/assets/941b2ec5-ec18-413f-9617-5a11fa503b4c" /> |
| <img src="https://github.com/user-attachments/assets/c09d1449-6f4e-438c-96b6-3e195547ae98" /> | <img src="https://github.com/user-attachments/assets/696cc8cd-cd61-49b5-8b44-46b57e806d3a" /> |

</p>
</details> 


## List of changes proposed in this pull request
- Added banner in `LandingScreen`
- Added banner in `MessageHomeScreen`
- Added banner in `LegacyMessageHomeScreen`

## How to test
- on iOS devices you'd see **NO** banner at all.
- on Android devices:
  - if the Android version is less than **7.1** you'd see the banner.
  - if the Android version is greater or equal to **7.1** you'd see NO banner at all.
